File: src/main/java/com/project/food_delivery/dtos/ProductMemoryValueData.java
package com.project.food_delivery.dtos;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.redis.core.RedisHash;

import java.io.Serializable;
import java.util.List;

@Data
@Builder
@RedisHash("Product")
@NoArgsConstructor
@AllArgsConstructor
public class ProductMemoryValueData implements Serializable {
    private String name;
    private ProductCategoryDto productCategory;
    private ProductCharacteristicDto productCharacteristic;
    private List<CompanyDtoRequest> companies;
    private PlaceDto place;
    private Integer amount;
}

File: front-end/src/app/common/product-memory-value-data.ts
import {ProductCategory} from "./product-category";
import {ProductCharacteristic} from "./product-characteristic";
import {CompanyDto} from "./company-dto";
import {Place} from "./place";

export class ProductMemoryValueData {
    public name: string;
    public productCategory: ProductCategory;
    public productCharacteristic: ProductCharacteristic;
    public companies: CompanyDto[];
    public place: Place;
    public amount: number;

    constructor(name: string, productCategory: ProductCategory, productCharacteristic: ProductCharacteristic, companies: CompanyDto[], place: Place, amount: number) {
        this.name = name;
        this.productCategory = productCategory;
        this.productCharacteristic = productCharacteristic;
        this.companies = companies;
        this.place = place;
        this.amount = amount;
    }
}
